### ISSUE TITLE
pre-push hook only checks files which contain the "vault" word in their name

### ISSUE THREAD

**Issue by ifraixedes** [cycloidio/cycloid-hooks#7]
The current `pre-push` hook only checks the files, whose name contains `vault`, for checking if the contain unencrypted data, however we are currently using other file names like [concourse-variables.yml](https://github.com/cycloidio/youdeploy-api/blob/5fd694ac9afef69c2f4f0c131ad8f18860d58994/ci/concourse-variables.yml)

We should be able to, at least, check the _concourse-varaible_ ones too

**Comment by xlr-8** [comment]
Yes, this is a limitation which is due to https://github.com/cycloidio/cycloid-hooks/commit/699afd4c461cc72ad15a5494127f1c87a75e5390, as it was based on https://github.com/cycloidio/cycloid-hooks/blob/master/client-side/pre-commit.d/pre-commit-ansible-vault that suffer from the same limitation.

Any PR are welcome :)

**Comment by ifraixedes** [comment]
According the comment https://github.com/cycloidio/cycloid-hooks/pull/8#pullrequestreview-20697678 we may think to change the strategy; rather than check the files name we could check the `yml` files which contains some stuff that we know that we must encrypt like `github-token`, `private-key` etc.

I don't know all the sensible data that we can identify in the `yml` files, but it looks more flexible to check the content than filter them by filename.

What do you think?

**Comment by xlr-8** [comment]
I'm actually wondering if we couldn't establish a list of file based on their "headers", as every encrypted file start with `$ANSIBLE_VAULT;1.1;AES256`. But that would require some deep checking of files & commits prior to push / commit via the hook.

**Comment by ifraixedes** [comment]
yes, but if the file is newly tracked it has never been encrypted before, so the hook will never catch it in that scenario.

**Comment by xlr-8** [comment]
Agreed, but that's nothing much we can do about it.

Checking the name variable is very unreliable and has too many
possibilities to be managed properly. e.g.: aws, vault, ansible, ssh-keys,
password, tokens, id, etc.
So I would assume to trust someone to push sensitive data encrypted at
least the first time, and then the hook would be able to know if that file
should remain encrypted.
